### PR TITLE
Switch to using openSUSE Leap 42.2

### LINKED ISSUE #144: Switch to openSUSE Leap
Leap is more stable than Tumbleweed.
Current problem with Tumbleweed is https://github.com/coala/docker-coala-base/issues/142 .

### PR DESCRIPTION
**Pull request by jayvdb**
Fixes https://github.com/coala/docker-coala-base/issues/144

### CODE REVIEW

**Comment by jayvdb** [comment]
This is not ideal at the moment.  I dont mind if we loose GoTypeBear, but JuliaLintBear is a huge loss.  Hopefully the upstream problems are solved quickly.

Noting that we've lost Julia, this removes ~180 Mb from the image, down to 2.017 GB.
Re-adding Julia may mean there are much smaller size improvements.
However, I also havent optimised this at all -- no doubt there are additional packages which can be removed after switching to Leap. (e.g. the custom Node could be replaced with the system Node.)

**Comment by jayvdb** [comment]
Note, this is close to finished, but there are a few things I would like to try.

**Comment by sils** [comment]
@jayvdb sounds like you don't want a full review yet? Taking a quick look and marking WIP after. Mark it pending review when done trying a few things.

**Comment by sils** [comment]
ack https://github.com/coala/docker-coala-base/pull/146/commits/699df2d3d0283813dc892b6c822f634a293c4137

File: Dockerfile [hunk 1]
```diff
@@ -305,10 +305,11 @@ RUN mkdir -p ~/.RLibrary && \
   find /tmp -mindepth 1 -prune -exec rm -rf '{}' '+'
 
 # Tailor (Swift) setup
+RUN curl -fsSL -o install.orig https://raw.githubusercontent.com/sleekbyte/tailor/master/script/install.sh
+
+RUN sed 's/read -r CONTINUE < \/dev\/tty/CONTINUE=y/' install.orig > install.sh
```
Comment (sils) [inline Dockerfile]: why move it up?

**Comment by jayvdb** [comment]
For some reason, go-1.5 is being selected.  Probably need to pin something.

**Comment by jayvdb** [comment]
Now busted by https://github.com/coala/docker-coala-base/issues/200

**Comment by sils** [comment]
> Now busted by #200

What do you mean?

**Comment by sils** [comment]
marking WIP because of conflicts and discussion at https://github.com/coala/docker-coala-base/issues/144#issuecomment-316653838

**Comment by jayvdb** [comment]
Current breakage is only https://github.com/coala/coala/issues/4519 , I believe.

**Review by jayvdb — COMMENTED** [review]
I think it would be good to put the base repo url and string "openSUSE_Leap_42.2" into env variables, to make these lines a bit shorter and less repetitive.

**Comment by fneu** [comment]
I don't think openSUSE Leap 42.2 is up to date anymore

**Pull request closed without merging**